scripts/pylib/twister/twisterlib.py

```diff
@@ -749,10 +749,19 @@ def handle(self):
         ser_pty_process = None
         if serial_pty:
             master, slave = pty.openpty()
+            serial_cmd = re.split(',| ', serial_pty)
+
+            # We also pass the build directory to the --device-serial-pty. In some boards, for example,
+            # intel_adsp_cavs series, their flash script and log capturing script might be put into
+            # one script. According to twister's working flow, the script is better to be put in
+            # --device-serial-pty script in these cases. So like what we pass to --west-flash script,
+            # we also pass also build directory to pty script, in order to get more flexibility.
```

Comment:
suggestion` # the pty script will ahve build directory as its first paramenter by derault, in order to get more flexibility.`

Reply:
```suggestion
            #  the pty script will ahve build directory as its first paramenter by derault, in order to get more flexibility.
```